lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -909,13 +987,19 @@ bool firrtl::areTypesConstCastable(FIRRTLType destFType, FIRRTLType srcFType,
   if (destFType == srcFType)
     return true;
 
-  auto destType = destFType.dyn_cast<FIRRTLBaseType>();
-  auto srcType = srcFType.dyn_cast<FIRRTLBaseType>();
+  auto destType =
+      firrtl::type_dyn_cast<FIRRTLBaseType>(destFType).getAnonymousType();
+  auto srcType =
+      firrtl::type_dyn_cast<FIRRTLBaseType>(srcFType).getAnonymousType();
 
   // For non-base types, only castable if identical.
   if (!destType || !srcType)
     return false;
 
+  // Equivalence is checked by their anonymous types.
+  destType = destType.getAnonymousType();
```

Comment:
Looks like the above `getAnonymousType()` should be removed in favor of unwrapping here after checking these are base types ?